pipeline_dp/pipeline_backend.py

```diff
@@ -474,11 +474,39 @@ def to_list(self, col, stage_name: str):
         raise NotImplementedError("to_list is not implement in SparkBackend.")
 
 
+class ReiterableLazyIterable(Iterable):
+    """A lazy iterable that can be iterated multiple times.
+
+    It generates elements on the first iteration and stores them.
+    Subsequent iterations yield the stored elements.
+    """
+
+    def __init__(self, iterable: Iterable):
+        """Initializes the ReiterableLazyIterable.
+
+        Args:
+            iterable: Iterable to make reiterable
+        """
+        self._iterable = iterable
+        self._cache: List = None
```

Comment:
why not to initialize it to `[]` immediately?

Reply:
self._cache = [] wouldn't work, because this list will shared between instances